src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -93,13 +97,39 @@ export const MultiSelect: ChakraComponent<
         ...rest
       } = props;
 
+      const [userClickedOutside, setUserClickedOutside] =
+        useState<boolean>(false);
+
       // Create a ref to hold a reference to the accordian button, enabling us
       // to programmatically focus it.
       const accordionButtonRef: React.RefObject<HTMLDivElement> =
         useRef<HTMLDivElement>();
+      const containerRef: React.RefObject<HTMLDivElement> =
+        useRef<HTMLDivElement>();
       const expandToggleButtonRef: React.RefObject<HTMLButtonElement> =
         useRef<HTMLButtonElement>();
 
+      const handleClick = (event) => {
```

Comment:
Chakra has a `useOutsideClick` function that sort of replicates what you have here. Did you get a chance to try that? Check out `useCloseDropDown.ts` for an example.

I think you can reuse the `useCloseDropDown` function here.